perllib/FixMyStreet/Cobrand/Bromley.pm

```diff
@@ -993,21 +994,72 @@ sub report_form_extras {
 
 =head2 Bulky waste collection
 
-Bromley has bulky waste collections starting at 6:30. It looks 4 weeks ahead
+Bromley has bulky waste collections starting at 7:00. It looks 4 weeks ahead
 for collection dates, and sends the event to the backend before collecting
 payment. Cancellations are done by update.
 
 =cut
 
-sub bulky_collection_time { { hours => 6, minutes => 30 } }
-sub bulky_cancellation_cutoff_time { { hours => 6, minutes => 30 } }
+sub bulky_collection_time { { hours => 7, minutes => 0 } }
+sub bulky_cancellation_cutoff_time { { hours => 7, minutes => 0 } }
 sub bulky_collection_window_days { 28 }
 sub bulky_cancel_by_update { 1 }
-sub bulky_can_refund { 0 }
 sub bulky_free_collection_available { 0 }
 sub bulky_hide_later_dates { 1 }
 sub bulky_send_before_payment { 1 }
 
+# TODO: Enforce the minumum charge.
+sub bulky_minimum_charge {
+    my $self = shift;
+    $self->feature('waste_features')->{bulky_minimum_charge};
+}
+
+sub bulky_can_refund_collection {
+    my ($self, $collection) = @_;
+    return 0 if !$self->within_bulky_cancel_window($collection);
+    return $self->bulky_refund_amount($collection) > 0;
+}
+
+sub bulky_refund_collection {
+    my ($self, $collection_report) = @_;
+    my $c = $self->{c};
+    $c->send_email(
+        'waste/bulky-refund-request.txt',
+        {   to => [
+                [ $c->cobrand->bulky_contact_email, $c->cobrand->council_name ]
+            ],
+
+            wasteworks_id => $collection_report->id,
+            payment_amount => $collection_report->get_extra_field_value('payment'),
+            payment_method =>
+                $collection_report->get_extra_field_value('payment_method'),
+            payment_code =>
+                $collection_report->get_extra_field_value('PaymentCode'),
+            auth_code =>
+                $collection_report->get_extra_metadata('authCode'),
+            continuous_audit_number =>
+                $collection_report->get_extra_metadata('continuousAuditNumber'),
+            payment_date       => $collection_report->created,
+            scp_response       =>
+                $collection_report->get_extra_metadata('scpReference'),
+        },
+    );
+}
+
+sub bulky_refund_would_be_partial {
+    my ($self, $collection) = @_;
+    my $d = $self->collection_date($collection);
+    my $t = $self->_bulky_cancellation_cutoff_date($d);
+    return $t->subtract( days => 1 ) <= DateTime->now;
+}
+
+sub bulky_refund_amount {
+    my ($self, $collection) = @_;
+    my $charged = $collection->get_extra_field_value('payment');
+    return ($charged - $self->bulky_minimum_charge) if $self->bulky_refund_would_be_partial($collection);
```

Comment:
I realise once minimum charge is in place this shouldn't be able to happen, but do we want to make sure this function can never return a negative amount, just in case? :)